### PR TITLE
rigsofrods: 0.4.7.0 -> 2021.04

### PR DESCRIPTION
**Pull request by luc65r**
###### Motivation for this change
RoR 0.4.7.0 is nearly 5 years old

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/mysocketw/default.nix [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
-    url = "https://www.digitalfanatics.org/cal/socketw/files/SocketW${version}.tar.gz";
-    sha256 = "0crinikhdl7xihzmc3k3k41pgxy16d5ci8m9sza1lbibns7pdwj4";
-  };
+  version = "3.10.26";
 
-  patches = [ ./gcc.patch ];
+  src = fetchFromGitHub {
+    owner = "RigsOfRods";
+    repo = "socketw";
+    rev = "f2094522b3940fd361ed9799bd0fcf4ba2b84988";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/mysocketw/default.nix:10]:
```suggestion
    rev = version;
```

File: pkgs/development/libraries/ogre/1.11.x.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{ fetchFromGitHub, stdenv, lib
+, cmake, libGLU, libGL
+, freetype, freeimage, zziplib, xorgproto, libXrandr
+, libXaw, freeglut, libXt, libpng, boost, ois
+, libX11, libXmu, libSM, pkg-config
+, libXxf86vm, libICE
+, libXrender
+, withNvidiaCg ? false, nvidia_cg_toolkit
+, withSamples ? false }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ogre/1.11.x.nix:9]:
```suggestion
, withSamples ? false
}:
```

File: pkgs/development/libraries/ogre/1.11.x.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  cmakeFlags = [ "-DOGRE_BUILD_SAMPLES=${toString withSamples}" ]
+    ++ map (x: "-DOGRE_BUILD_PLUGIN_${x}=on")
+           ([ "BSP" "OCTREE" "PCZ" "PFX" ] ++ lib.optional withNvidiaCg "CG")
+    ++ map (x: "-DOGRE_BUILD_RENDERSYSTEM_${x}=on") [ "GL" ];
+
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs =
+   [ libGLU libGL
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ogre/1.11.x.nix:30]:
```suggestion
  buildInputs = [
    libGLU libGL
```

File: pkgs/development/libraries/ogre/1.11.x.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+     libXxf86vm libICE
+     libXrender
+   ] ++ lib.optional withNvidiaCg nvidia_cg_toolkit;
+
+  meta = {
+    description = "A 3D engine";
+    homepage = "https://www.ogre3d.org/";
+    maintainers = [ lib.maintainers.raskin ];
+    platforms = lib.platforms.linux;
+    license = lib.licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ogre/1.11.x.nix:43]:
```suggestion
  meta = with lib; {
    description = "A 3D engine";
    homepage = "https://www.ogre3d.org/";
    maintainers = [ maintainers.raskin ];
    platforms = platforms.linux;
    license = licenses.mit;
```
Also please only add yourself as a maintainer.

File: pkgs/development/libraries/ogre/default.nix [hunk 5]
```diff
@@ -7,19 +7,39 @@
...
 
-  cmakeFlags = [ "-DOGRE_BUILD_DEPENDENCIES=OFF" "-DOGRE_BUILD_SAMPLES=${toString withSamples}" ]
+  preConfigure = lib.optionalString withImgui ''
+    mkdir build
+    cp -r ${imgui} build/imgui-1.79
+    chmod u+w -R build
+  '';
+
+  cmakeFlags = [ "-DOGRE_BUILD_DEPENDENCIES=OFF" "-DOGRE_BUILD_SAMPLES=${toString withSamples}"
+                  ]
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ogre/default.nix:42]:
```suggestion
  cmakeFlags = [ "-DOGRE_BUILD_DEPENDENCIES=OFF" "-DOGRE_BUILD_SAMPLES=${toString withSamples}" ]
```

File: pkgs/development/libraries/ogre/default.nix [hunk 6]
```diff
@@ -34,12 +54,13 @@ stdenv.mkDerivation rec {
...
      libXrender
      SDL2
+     pugixml
    ] ++ lib.optional withNvidiaCg nvidia_cg_toolkit;
 
   meta = {
     description = "A 3D engine";
     homepage = "https://www.ogre3d.org/";
-    maintainers = [ lib.maintainers.raskin ];
+    maintainers = [ lib.maintainers.raskin lib.maintainers.luc65r ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ogre/default.nix:63]:
```suggestion
    maintainers = with maintainers; [ raskin luc65r ];
```

File: pkgs/development/libraries/ogre/default.nix [hunk 7]
```diff
@@ -34,12 +54,13 @@ stdenv.mkDerivation rec {
      libXxf86vm libICE
      libXrender
      SDL2
+     pugixml
    ] ++ lib.optional withNvidiaCg nvidia_cg_toolkit;
 
   meta = {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ogre/default.nix:60]:
```suggestion
  meta = with lib; {
```

File: pkgs/development/libraries/ogre/default.nix [hunk 8]
```diff
@@ -34,12 +54,13 @@ stdenv.mkDerivation rec {
...
     description = "A 3D engine";
     homepage = "https://www.ogre3d.org/";
-    maintainers = [ lib.maintainers.raskin ];
+    maintainers = [ lib.maintainers.raskin lib.maintainers.luc65r ];
     platforms = lib.platforms.linux;
     license = lib.licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ogre/default.nix:65]:
```suggestion
    platforms = platforms.linux;
    license = licenses.mit;
```

File: pkgs/development/libraries/ogre/default.nix (hunk 7 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/ogre/default.nix:60]: If we are redoing large parts of the package anyway we can also clean this up.

Comment (jljusten) [reply pkgs/development/libraries/ogre/default.nix:60]: @luc65r: Sounds like you might have some bigger rework plans, but I put some "fixup" patches based on @SuperSandro2000's suggestions into this branch. I also had to bump the mysocketw version to 3.10.27 to get it to build with openssl.

https://github.com/jljusten/nixpkgs/commits/rigsofrods-2021.04

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**